src/server/routes/contract/events/get-all-events.ts

```diff
@@ -82,20 +85,67 @@ export async function getAllEvents(fastify: FastifyInstance) {
       const { fromBlock, toBlock, order } = request.query;
 
       const chainId = await getChainIdFromChain(chain);
-      const contract = await getContract({
-        chainId,
-        contractAddress,
-      });
 
-      let returnData = await contract.events.getAllEvents({
-        fromBlock,
-        toBlock,
-        order,
+      const contract = getContract({
+        client: thirdwebClient,
+        address: contractAddress,
+        chain: await getChain(chainId),
       });
+      const returnData = mapEventsV4ToV5(
+        await getContractEvents({
+          contract: contract,
+          fromBlock: BigInt(fromBlock),
+          toBlock: BigInt(toBlock),
+        }),
+        order,
+      );
 
       reply.status(StatusCodes.OK).send({
         result: returnData,
       });
     },
   });
 }
+
+/**
+ * Mapping of events v5 response to v4 for backward compatiblity.
+ * Clients may be using this api and dont want to break things.
+ * @param eventsV5 events data returned by v5 sdk
+ * @param order asc or desc
+ * @returns {Type.Array(contractEventSchema)}
+ */
+export function mapEventsV4ToV5(eventsV5 = [], order = "desc") {
+  if (!eventsV5?.length) return [];
+
+  return eventsV5
+    .map((event) => {
+      const eventName = event.eventName;
+      const data = {};
+
+      // backwards compatibility of BigInt(v5) to BigNumber(v4)
+      Object.keys(event.args).forEach((key) => {
+        let value = event.args[key];
+        if (typeof value == "bigint") {
+          value = BigNumber.from(value.toString());
+        }
+        data[key] = value;
+      });
+
+      delete event.eventName;
```

Comment:
It's less readable and performant to delete fields. It's more readable to explicitly map fields from one object to another, even if it's long

```
const fooV5 = {
  field1: v4.field1,
  field2: v4.field2,
  field3: BigNumber.from(v4.field3),
  ...
}
```

Reply:
But if the library updates or the data structure adds new fields, we'll have to remember to add it here. Though typescript will probably give error for that as well and not build so might as well map each fields manually :+1: